ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransactions.java

```diff
@@ -331,6 +331,11 @@ public OptionalLong getNextNonceForSender(final Address sender) {
       if (transactionsForSender == null) {
         return OptionalLong.empty();
       }
+      for (final Long nonce : transactionsForSender.keySet()) {
```

Comment:
I'm concerned about how this would perform under performance testing load.  This addresses the immediate problem but introduces performance debt.  This loop will get called a lot.  

Could we replace the Map<Address, SortedMap<Long, TransactionInfo>> with a different structure?  Perhaps instead of a Sorted map a new object that has the infos in a nonce-sorted list and then a sorted list of "gaps" (TreeSet, or maybe a PriorityQueue) and when queried we just peek the lowest number and return that, if there are no gaps then we peek the nonce of the last transaction info and return that value, and if empty query the account as a last resort.  We would need to add logic for when we add and remove transactions to update the gap list.

Reply:
I understand your concern. You are right. I will work a better implementation.

Reply:
Fixed https://github.com/hyperledger/besu/pull/351/commits/4e6b11e76260cca09eec9a168f2440c0dbffd18d